Services/Simulations.cs

```diff
@@ -293,6 +319,18 @@ public async Task AddDeviceAsync(string id)
             await this.storage.CreateAsync(DEVICES_COLLECTION, id, id);
         }
 
+        /// <summary>
+        /// Update simulation definition in storage.
+        /// </summary>
+        public async Task UpdateStorage(Models.Simulation simulation)
```

Comment:
could we use `UpsertAsync` instead of this method? looks like they do the same thing, no? "Upsert" is used for creation and update, so it should work

Reply:
:white_check_mark: